run.sh

```diff
@@ -77,6 +85,16 @@ do
           fi
         done
       fi
+      if [ -n "${KEEP_CONTAINERS_NAMED}" ]; then
```

Comment:
A bit of duplication with this block here, perhaps something that could be refactored out as a bash function?